device_operation/include/device_reduce_blockwise.hpp

```diff
@@ -0,0 +1,350 @@
+#ifndef DEVICE_REDUCE_BLOCKWISE_HPP
+#define DEVICE_REDUCE_BLOCKWISE_HPP
+
+#include <iostream>
+#include <sstream>
+#include "device.hpp"
+#include "device_reduce.hpp"
+#include "device_reduce_common.hpp"
+#include "gridwise_2d_reduction_blockwise.hpp"
+
+namespace ck {
+namespace tensor_operation {
+namespace device {
+
+template <typename InDataType,
+          typename AccDataType,
+          typename OutDataType,
+          int Rank,
+          typename InnerDims,
+          typename ReduceOperation,
+          typename InElementwiseOperation,
+          typename AccElementwiseOperation,
+          bool PropagateNan,
+          bool NeedIndices,
+          int BlockSize,
```

Comment:
please use index_t, because it will be passed to kernel code as template argument

Reply:
Ok.   So when I need to use int32_t, for example for some sizes, should I also use index_t ?  I think for int32_t,  we need use  namespace std since it is defined by c++

Reply:
int32_t is used only for tensor value

index_t is used for anything related to addresr, tensor coordinate, length, stec